src/main/java/com/eprosima/idl/parser/tree/Param.java

```diff
@@ -106,9 +112,26 @@ public Object getParent()
         return m_parent;
     }
 
+    @Override
+    public void addAnnotation(Context ctx, Annotation annotation)
+    {
+        if(annotation != null)
+        {
+            m_annotations.put(annotation.getName(), annotation);
+        }
+    }
+
+    @Override
+    public Map<String, Annotation> getAnnotations()
+    {
+        return m_annotations;
+    }
+
     private Kind m_kind = Kind.IN_PARAM;
     private String m_name = null;
     private TypeCode m_typecode = null;
     private Definition m_definition = null;
     private Object m_parent = null;
+    //! Map that stores the annotations of the parameter.
+    private HashMap<String, Annotation> m_annotations = null;
```

Comment:
In the last years I was changing the initialization of class' members from the constructor to the declaration. I suggest:

```suggestion
    private HashMap<String, Annotation> m_annotations = new HashMap<String, Annotation>();
```